### PR TITLE
Wizard recipe: YASM-v1.3.0

### PR DESCRIPTION
**Pull request by innerlee**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: YASM
* Version: v1.3.0

@staticfloat please review and merge.

### CODE REVIEW

**Comment by innerlee** [comment]
I don't know how to solve the error 

image

**Comment by giordano** [comment]
I'm writing down here the error messages for easier discoverability (a screenshot isn't very useful):
```
[06:49:21] /opt/bin/x86_64-linux-musl-cxx11/x86_64-linux-musl-gcc -I.  -c -o genversion.o `test -f modules/preprocs/nasm/genversion.c || echo './'`modules/preprocs/nasm/genversion.c
[06:49:21] /opt/bin/x86_64-linux-musl-cxx11/x86_64-linux-musl-gcc -I.  -c -o genstring.o `test -f genstring.c || echo './'`genstring.c
[06:49:21] cc -o genperf  genperf.o gp-perfect.o gp-phash.o gp-xmalloc.o gp-xstrdup.o 
[06:49:21] cc -o genversion  genversion.o 
[06:49:21] cc -o genmacro  genmacro.o 
[06:49:21] cc -o genstring  genstring.o 
[06:49:21] cc -o re2c  re2c-main.o re2c-code.o re2c-dfa.o re2c-parser.o re2c-actions.o re2c-scanner.o re2c-mbo_getopt.o re2c-substr.o re2c-translate.o 
[06:49:21] /opt/i686-linux-gnu/bin/../lib/gcc/i686-linux-gnu/4.8.5/../../../../i686-linux-gnu/bin/ld: i386:x86-64 architecture of input file `genperf.o' is incompatible with i386 output
[06:49:21] /opt/i686-linux-gnu/bin/../lib/gcc/i686-linux-gnu/4.8.5/../../../../i686-linux-gnu/bin/ld: i386:x86-64 architecture of input file `gp-perfect.o' is incompatible with i386 output
[06:49:21] /opt/i686-linux-gnu/bin/../lib/gcc/i686-linux-gnu/4.8.5/../../../../i686-linux-gnu/bin/ld: i386:x86-64 architecture of input file `gp-phash.o' is incompatible with i386 output
[06:49:21] /opt/i686-linux-gnu/bin/../lib/gcc/i686-linux-gnu/4.8.5/../../../../i686-linux-gnu/bin/ld: i386:x86-64 architecture of input file `gp-xmalloc.o' is incompatible with i386 output
[06:49:21] /opt/i686-linux-gnu/bin/../lib/gcc/i686-linux-gnu/4.8.5/../../../../i686-linux-gnu/bin/ld: i386:x86-64 architecture of input file `gp-xstrdup.o' is incompatible with i386 output
[06:49:21] /opt/i686-linux-gnu/bin/../lib/gcc/i686-linux-gnu/4.8.5/../../../../i686-linux-gnu/bin/ld: genperf.o: file class ELFCLASS64 incompatible with ELFCLASS32
[06:49:21] /opt/i686-linux-gnu/bin/../lib/gcc/i686-linux-gnu/4.8.5/../../../../i686-linux-gnu/bin/ld: final link failed: File in wrong format
[06:49:21] collect2: error: ld returned 1 exit status
[06:49:21] make: *** [Makefile:2692: genperf] Error 1
```
I haven't seen the build system, but it looks like a mess, it's mixing object compiled with the host compiler with those compiled with the target compiler.

File: Y/YASM/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("http://www.tortall.net/projects/yasm/releases/yasm-1.3.0.tar.gz", "3dce6601b495f5b3d45b59f7d2492a340ee7e84b5beca17e48f862502bd5603f")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/yasm-*
+autoreconf -f -i
```
Comment (giordano) [inline Y/YASM/build_tarballs.jl:16]:
```suggestion
autoreconf -f -i
export CCLD_FOR_BUILD="${CC_FOR_BUILD}"
```

File: Y/YASM/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+make -j${nproc}
+make install
+install_license COPYING
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux")
+]
```
Comment (giordano) [inline Y/YASM/build_tarballs.jl:27]:
```suggestion
platforms = supported_platforms(; experimental=true)
```

File: Y/YASM/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    ExecutableProduct("vsyasm", :vsyasm)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="NASM_jll", uuid="08ca2550-6d73-57c0-8625-9b24120f3eae"))
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```
Comment (giordano) [inline Y/YASM/build_tarballs.jl:42]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```

**Comment by innerlee** [comment]
amazing

**Pull request merged**